in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,265 @@
+- `BLOBS_PER_BLOCK` - the number of blobs per block that are compensated for in the base fee (e.g. 3)
+- `L2_SLOTS_PER_L2_EPOCH` - the number of L2 slots in each L2 epoch (e.g. 32)
+- `L1_GAS_PER_BLOCK_PROPOSED` - the amount of L1 gas required to propose an L2 block on L1 (e.g. 0.2e6)
+- `L1_GAS_PER_EPOCH_VERIFIED` - the amount of L1 gas required to verify an L2 epoch on L1 (e.g. 1e6)
+- `MINIMUM_PROVING_COST_PER_MANA` - the minimum cost in wei for proving a unit of mana (e.g. 100)
+- `MAXIMUM_PROVING_COST_PERCENT_CHANGE_PER_BLOCK` - the maximum percentage increase in the cost of proving a unit of mana per block (e.g. 1%)
+- `PROVING_COST_PRECISION` - the precision of the `proving_cost_modifier` (e.g. 1e9)
+- `MINIMUM_FEE_ASSET_PRICE` - the minimum price in wei of the fee asset (e.g. TST)
+- `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` - the maximum percentage increase in the price of the fee asset per block (e.g. 1%)
+- `FEE_ASSET_PRICE_PRECISION` - the precision of the `fee_asset_price_modifier` (e.g. 1e9)
+- `UNDERLYING_BASE_FEE_ORACLE_UPDATE_INTERVAL` - the minimum number of slots between updates to the underlying base fee oracle (e.g. 4)
+- `MINIMUM_CONGESTION_MULTIPLIER` - the minimum value the congestion multiplier can take (e.g. 1)
+- `CONGESTION_MULTIPLIER_DAMPER` - the constant factor to dampen movement in the congestion multiplier (e.g. 8.547)
+- `MAXIMUM_EPOCH_PROOF_QUOTE_FEE` - the maximum basis point fee a prover can submit for an epoch proof quote (e.g. 9000)
+
+## Block Header Fields
+
+- `total_mana_used` - the total mana used by the block
+- `excess_mana` - the running total of excess mana on the chain relative to `TARGET_MANA_PER_BLOCK`
+- `base_fee_asset_per_mana` - the base fee in fee asset per mana
+
+## Rollup Contract Fields
+
+- `proving_cost_modifier` - a value used in the computation of the proving cost per mana
+- `proving_cost_modifier_delta` - an "oracle" used to update the `proving_cost_modifier` from the previous block
+- `fee_asset_price_modifier` - a value used in the computation of the fee asset price per wei
+- `fee_asset_price_modifier_delta` - an "oracle" used to update the `fee_asset_price_modifier` from the previous block
+- `wei_per_l1_gas` - the cost of L1 gas in wei. Updated by anyone to the current L1 gas price at most every `UNDERLYING_BASE_FEE_ORACLE_UPDATE_INTERVAL` slots
+- `wei_per_l1_blob_gas` - the cost of L1 blob gas in wei. Updated by anyone to the current L1 blob gas price at most every `UNDERLYING_BASE_FEE_ORACLE_UPDATE_INTERVAL` slots
+
+## Block Execution Cost
+
+The cost of a block's execution in wei is equal to the following:
+```math
+\begin{aligned}
+\text{execution gas} &= \text{L1 GAS PER BLOCK PROPOSED}  \\
+&+ \text{BLOBS PER BLOCK} * \text{POINT EVALUATION PRECOMPILE GAS} \\
+&+ \frac{\text{L1 GAS PER EPOCH VERIFIED}}{\text{L2 SLOTS PER L2 EPOCH}} \\
+\text{execution wei} &= \text{execution gas} * \text{wei per l1 gas}
```

Comment:
Down here, I would rather call it `execution_cost` to make it clearer that it is a cost. Kinda like, having a execution cost measured in usd, it is not execution_usd. Kinda a nit, but just clarity. Probably don't really need to specify the `execution_gas` can just go directly to the cost. Or, l1 costs at least.

Reply:
I've made it very clear. I also kept the additional step to help with formatting.